coffea/nanoevents/schemas/delphes.py

```diff
@@ -0,0 +1,290 @@
+        "SoftDroppedP4[5]": "first entry (i = 0) is the total SoftDropped Jet 4-momenta and from i = 1 to 4 are the pruned subjets 4-momenta",
+        "Status": "particle status",
+        "SumPT2": "sum pt^2 of tracks attached to the vertex",
+        "SumPt": "isolation variable",
+        "SumPtCharged": "isolation variable",
+        "SumPtChargedPU": "isolation variable",
+        "SumPtNeutral": "isolation variable",
+        "TOuter": "time position (t component) at the edge",
+        "TauTag": "0 or 1 for a particle that has been tagged as a tau",
+        "Tau[5]": "N-subjettiness",
+        "Trigger": "trigger word",
+        "TrimmedP4[5]": "first entry (i = 0) is the total Trimmed Jet 4-momenta and from i = 1 to 4 are the trimmed subjets 4-momenta",
+        "Tx": "angle of the momentum in the horizontal (x,z) plane [urad]",
+        "Ty": "angle of the momentum in the verical (y,z) plane [urad]",
+        "VertexIndex": "reference to vertex",
+        "Weight": "weight for the event",
+        "X1": 'fraction of beam momentum carried by first parton ("beam side")',
+        "X2": 'fraction of beam momentum carried by second parton ("target side")',
+        "XOuter": "position (x component) at the edge",
+        "Xd": "X coordinate of point of closest approach to vertex",
+        "YOuter": "position (y component) at the edge",
+        "Yd": "Y coordinate of point of closest approach to vertex",
+        "ZOuter": "position (z component) at the edge",
+        "Zd": "Z coordinate of point of closest approach to vertex",
+    }
+
+    def __init__(self, base_form, version="latest"):
+        super().__init__(base_form)
+        self._version = version
+        if version == "latest":
+            pass
+        else:
+            pass
+        self._form["contents"] = self._build_collections(self._form["contents"])
+        self._form["parameters"]["metadata"]["version"] = self._version
+
+    @classmethod
+    def v1(cls, base_form):
+        """Build the DelphesEvents
+
+        For example, one can use ``NanoEventsFactory.from_root("file.root", schemaclass=DelphesSchema.v1)``
+        to ensure NanoAODv7 compatibility.
+        """
+        return cls(base_form, version="1")
+
+    def _build_collections(self, branch_forms):
+        # parse into high-level records (collections, list collections, and singletons)
+        collections = set(k.split("/")[0] for k in branch_forms)
+        collections -= set(k for k in collections if k.endswith("_size"))
+
+        # Create offsets virtual arrays
+        for name in collections:
+            if f"{name}_size" in branch_forms:
+                branch_forms[f"o{name}"] = transforms.counts2offsets_form(
+                    branch_forms[f"{name}_size"]
+                )
+
+        for name in collections:
+            for key in self.tlorentzvectors:
+                if f"{name}/{name}.{key}" in branch_forms:
+                    ...  # do stuff
```

Comment:
I imagine we need to do something for any of the TLV cases, once they've been defined properly in `uproot/mapping.py` and "transform" them accordingly.

Reply:
These are actually uproot fault, so I'll fix upstream